frontend/src/store/user/types.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  FETCH_USER_REQUEST,
+  FETCH_USER_SUCCESS,
+  FETCH_USER_FAILURE,
+  REGISTER_USER_REQUEST,
+  REGISTER_USER_SUCCESS,
+  REGISTER_USER_FAILURE,
+} from "./actionTypes";
+
+export interface IUser {
+  token: string;
+}
+
+export interface UserState {
+  pending: boolean;
+  user: IUser | null;
+  error: string | null;
+  registrationSuccess: boolean | null;
+}
+
+export interface FetchUserPayload {
+  login: string;
+  password: string;
+}
+
+export interface FetchUserSuccessPayload {
+  user: IUser;
+}
+
+export interface FetchUserFailurePayload {
+  error: string;
+}
+
+export interface RegisterUserPayload {
+  mail: string;
```

Comment:
Better to have the same keys for username in login/mail in log in and register case